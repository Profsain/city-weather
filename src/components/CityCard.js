import PropTypes from "prop-types"
import React from 'react';
import '../styles/CityCard.css';

const CityCard = ({ cityName, cityPic }) => (
  <div className="City-card">
    <img src={cityPic} alt="abuja city" />
    <h4>{cityName}</h4>
  </div>
);

CityCard.propTypes = {
  cityName: PropTypes.string.isRequired,
  cityPic: PropTypes.string.isRequired,
};

export default CityCard;
