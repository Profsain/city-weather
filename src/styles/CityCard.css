.City-card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 1px solid #fff;
  border-radius: 3px;
  padding: 18px;
}

.City-card img {
  width: 180px;
  height: 150px;
}

.City-card:hover {
  cursor: pointer;
  border: 2px solid rgb(136, 136, 136);
}
