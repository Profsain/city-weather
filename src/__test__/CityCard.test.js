import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configurestore';
import CityCard from '../components/CityCard';

it('Renders CityCard component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CityCard />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
