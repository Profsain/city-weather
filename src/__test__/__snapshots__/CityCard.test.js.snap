// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CityCard component correctly 1`] = `
<div
  className="City-card"
>
  <img
    alt="abuja city"
  />
  <h4 />
</div>
`;
